lib/OpenQA/Worker/Cache/Service.pm

```diff
@@ -0,0 +1,206 @@
+# Copyright (C) 2018 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package OpenQA::Worker::Cache::Service;
+
+use Mojolicious::Lite;
+use OpenQA::Worker::Cache::Task::Asset;
+use Mojolicious::Plugin::Minion;
+use OpenQA::Worker::Cache qw(STATUS_PROCESSED STATUS_ENQUEUED STATUS_DOWNLOADING STATUS_IGNORE STATUS_ERROR);
+use Mojo::Collection;
+use Mojolicious::Plugin::Minion::Admin;
+
+use constant DEFAULT_MINION_WORKERS => 5;
+
+BEGIN { srand(time) }
+
+my $_token;
+
+sub _gen_session_token { $_token = int(rand(999999999999)) }
```

Comment:
Maybe include `$$` and `time` to make it more unique?

Reply:
My default is usually `md5_sum($$ . time . rand)`.

Reply:
Done in https://github.com/os-autoinst/openQA/pull/1783/commits/71e142c959efcf6c2e395f891e7522869be78d7d